lib/views/chatroom/chatroom_components/chat_bubble.dart

```diff
@@ -0,0 +1,367 @@
+                                  } else {
+                                    return const Spinner();
+                                  }
+                                }),
+                                future: getContent(
+                                    _contentType, _message, _isSent)),
+                            const SizedBox(height: 8),
+                            Text(
+                              _time,
+                              style: GoogleFonts.roboto(
+                                fontSize: 12,
+                                color: Colors.black.withOpacity(0.6),
+                              ),
+                            ),
+                          ],
+                        ),
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+              const SizedBox(width: 6),
+              _isSent
+                  ? Container(
+                      width: 42,
+                      height: 42,
+                      decoration: BoxDecoration(
+                        color: Colors.blue,
+                        borderRadius: BorderRadius.circular(21),
+                        image: _profileImageUrl != null
+                            ? DecorationImage(
+                                image: NetworkImage(_profileImageUrl),
+                                fit: BoxFit.cover,
+                              )
+                            : null,
+                      ),
+                    )
+                  : const SizedBox(),
+            ],
+          ),
+        ),
+        if (reactions.isNotEmpty)
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 64.0),
+            child: Row(
+              mainAxisAlignment:
+                  _isSent ? MainAxisAlignment.end : MainAxisAlignment.start,
+              children: reactions.map((e) {
+                switch (e) {
+                  case Reactions.like:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":thumbsup:"),
+                    );
+                  case Reactions.dislike:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":thumbsdown:"),
+                    );
+                  case Reactions.love:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":heart:"),
+                    );
+                  case Reactions.happy:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":joy:"),
+                    );
+                  case Reactions.sad:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":cry:"),
+                    );
```

Comment:
It will be better to breakdown this factory method to return the ReactionChip to another function.